data-pipeline-flink/assessment-aggregator/src/main/scala/org/sunbird/dp/assessment/functions/AssessmentAggregatorFunction.scala

```diff
@@ -39,125 +40,139 @@ class AssessmentAggregatorFunction(config: AssessmentAggregatorConfig,
-          .from(config.dbKeyspace, config.dbTable).
-          where(QueryBuilder.eq("attempt_id", event.attemptId))
-          .and(QueryBuilder.eq("batch_id", event.batchId))
-          .and(QueryBuilder.eq("user_id", event.userId))
-          .and(QueryBuilder.eq("content_id", event.contentId))
-          .and(QueryBuilder.eq("course_id", event.courseId)).toString
-
-        cassandraUtil.findOne(query)
-
-    }
-
-
-    def saveAssessment(batchEvent: Event, aggregate: Aggregate, createdOn: Long): Unit = {
-        val query = QueryBuilder.insertInto(config.dbKeyspace, config.dbTable)
-          .value("course_id", batchEvent.courseId).value("batch_id", batchEvent.batchId).value("user_id", batchEvent.userId)
-          .value("content_id", batchEvent.contentId).value("attempt_id", batchEvent.attemptId)
-          .value("updated_on", new DateTime(DateTimeZone.UTC).getMillis).value("created_on", createdOn)
-          .value("last_attempted_on", batchEvent.assessmentEts).value("total_score", aggregate.totalScore)
-          .value("total_max_score", aggregate.totalMaxScore)
-          .value("question", aggregate.questionsList.asJava).value("grand_total", aggregate.grandTotal).toString
-
-        cassandraUtil.upsert(query)
-        logger.info("Successfully Aggregated the batch event - batchid: "
-          + batchEvent.batchId + " ,userid: " + batchEvent.userId + " ,couserid: "
-          + batchEvent.courseId + " ,contentid: " + batchEvent.contentId, "attempid" + batchEvent.attemptId)
-    }
-
-    def getQuestion(questionData: QuestionData, assessTs: Long): UDTValue =
-
-        questionType.newValue().setString("id", questionData.item.id).setDouble("max_score", questionData.item.maxscore)
-          .setDouble("score", questionData.score)
-          .setString("type", questionData.item.`type`)
-          .setString("title", questionData.item.title)
-          .setList("resvalues", getListValues(questionData.resvalues).asJava).setList("params", getListValues(questionData.item.params).asJava)
-          .setString("description", questionData.item.desc)
-          .setDecimal("duration", BigDecimal.valueOf(questionData.duration)).setTimestamp("assess_ts", new Timestamp(assessTs))
+  }
+
+  def getAssessment(event: Event): Row = {
+
+    val query = QueryBuilder.select("last_attempted_on", "created_on")
+      .from(config.dbKeyspace, config.dbTable).
+      where(QueryBuilder.eq("attempt_id", event.attemptId))
+      .and(QueryBuilder.eq("batch_id", event.batchId))
+      .and(QueryBuilder.eq("user_id", event.userId))
+      .and(QueryBuilder.eq("content_id", event.contentId))
+      .and(QueryBuilder.eq("course_id", event.courseId)).toString
+
+    cassandraUtil.findOne(query)
+
+  }
+
+  def isValidContent(courseId: String, contentId: String)(metrics: Metrics): Boolean = {
+    val leafNodes = dataCache.getKeyMembers(s"$courseId:$courseId:leafnodes")
+    metrics.incCounter(config.cacheHitCount)
+    leafNodes.contains(contentId)
```

Comment:
@maheshkumargangula @manjudr - the logic should be backward compatible. What if the cache hit is a miss? This should return true for the following cases:
1. if the cache entry doesn't exist or the leafnodes are empty
2. if the leafnodes are not empty and the contentId exists in the leafnode